pkg/forklift-api/webhooks/validating-webhook/admitters/storage-admitter.go

```diff
@@ -0,0 +1,125 @@
+package admitters
+
+import (
+	"context"
+	v1 "k8s.io/api/storage/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"encoding/json"
+	"fmt"
+	admissionv1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"net/http"
+
+	"github.com/konveyor/forklift-controller/pkg/apis"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
+)
+
+type StorageAdmitter struct {
+}
+
+func (admitter *StorageAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	log.Info("storage admitter was called")
+	raw := ar.Request.Object.Raw
+
+	if len(raw) == 0 {
+		// plan is being deleted, passing
+		return &admissionv1.AdmissionResponse{
+			Allowed: true,
+		}
+	}
+
+	plan := &api.Plan{}
+	err := json.Unmarshal(raw, plan)
+	if err != nil {
+		return util.ToAdmissionResponseError(err)
+	}
+
+	if plan.Spec.Warm {
+		log.Info("Warm migration supports all storages, passing")
+		return &admissionv1.AdmissionResponse{
+			Allowed: true,
+		}
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Info("Couldn't get the cluster configuration", "error", err.Error())
+	}
+
+	err = api.SchemeBuilder.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Info("Couldn't build the scheme", "error", err.Error())
+	}
+	err = apis.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Info("Couldn't add forklift API to the scheme", "error", err.Error())
+	}
+
+	cl, err := client.New(config, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		log.Info("Couldn't create a cluster client", "error", err.Error())
+	}
+
+	sourceProvider := api.Provider{}
+	err = cl.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Source.Namespace, Name: plan.Spec.Provider.Source.Name}, &sourceProvider)
+	if err != nil {
+		log.Info("Couldn't get the source provider", "error", err.Error())
```

Comment:
actually, maybe we should allow it when getting to such error. otherwise we will block any creation/edit of a plan if an error occurred in the webhook. i think it's bad to block it in such case (better to have migrations working, as no-provisioner is anyway not commonly used). WDYT?

Reply:
yes it makes sense not to block the plan creation over these errors, so in that case you should return an allowed request but mention in the log and message that we had an error, something like in this block [https://github.com/kubev2v/forklift/blob/cf2ad9146d2c17e5a88668fe73a27563e29fa2ce/pkg/forklift-api/webhooks/validating-webhook/admitters/secret-admitter.go#L72] , but we must have some return statement, otherwise it will cause the API pod to crash (it will try to query info that wasn't retrieved properly) and we don't want it to happen.

Reply:
also because you will need to return this in a lot of places of the code you can add a util function for this purpose [here](pkg/forklift-api/webhooks/util/util.go) to make it more readable if you like .

Reply:
why would this request fail though? if the plan is going to be successfully executed these requests should work regardless